optimize/backend/src/main/java/io/camunda/optimize/TomcatConfig.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize;
+
+import static io.camunda.optimize.service.util.configuration.EnvironmentPropertiesConstants.CONTEXT_PATH;
+
+import io.camunda.optimize.rest.HealthRestService;
+import io.camunda.optimize.rest.LocalizationRestService;
+import io.camunda.optimize.rest.UIConfigurationRestService;
+import io.camunda.optimize.rest.constants.RestConstants;
+import io.camunda.optimize.rest.security.cloud.CCSaasAuth0WebSecurityConfig;
+import io.camunda.optimize.service.exceptions.OptimizeConfigurationException;
+import io.camunda.optimize.service.util.PanelNotificationConstants;
+import io.camunda.optimize.service.util.configuration.ConfigurationService;
+import io.camunda.optimize.service.util.configuration.EnvironmentPropertiesConstants;
+import io.camunda.optimize.tomcat.ExternalHomeServlet;
+import io.camunda.optimize.tomcat.OptimizeResourceConstants;
+import io.camunda.optimize.tomcat.ResponseHeadersFilter;
+import io.camunda.optimize.tomcat.URLRedirectFilter;
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRegistration;
+import java.net.URL;
+import java.util.Optional;
+import org.apache.catalina.connector.Connector;
+import org.apache.coyote.http2.Http2Protocol;
+import org.apache.tomcat.util.net.SSLHostConfig;
+import org.apache.tomcat.util.net.SSLHostConfigCertificate;
+import org.apache.tomcat.util.net.SSLHostConfigCertificate.Type;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.server.WebServerFactoryCustomizer;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.boot.web.servlet.ServletContextInitializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+@Configuration
+public class TomcatConfig {
```

Comment:
NABD but could call this `OptimizeTomcatConfig` just for specificity, given tomcat is used broadly